sw/device/silicon_creator/lib/drivers/keymgr_functest.c

```diff
@@ -235,22 +150,24 @@ static void init_kmac_for_keymgr(void) {
       .sideload = true,
   };
   CHECK_DIF_OK(dif_kmac_configure(&kmac, config));
+  for (size_t i = 0; i < kKmacPrefixSize; ++i) {
+    mmio_region_write32(mmio_region_from_addr(TOP_EARLGREY_KMAC_BASE_ADDR),
+                        KMAC_PREFIX_0_REG_OFFSET + i * 4, kKmacPrefix[i]);
+  }
 }
 
-/** Soft reboot device. */
-static void soft_reboot(dif_pwrmgr_t *pwrmgr, dif_aon_timer_t *aon_timer) {
-  aon_timer_wakeup_config(aon_timer, 1);
-
-  // Place device into low power and immediately wake.
-  dif_pwrmgr_domain_config_t config;
-  config = kDifPwrmgrDomainOptionUsbClockInActivePower;
-
-  pwrmgr_testutils_enable_low_power(pwrmgr, kDifPwrmgrWakeupRequestSourceFive,
-                                    config);
-
-  // Enter low power mode.
-  LOG_INFO("Entering low power");
-  wait_for_interrupt();
+static void check_lock_otp_partition(const dif_otp_ctrl_t *otp) {
+  bool is_computed;
+  CHECK_DIF_OK(dif_otp_ctrl_is_digest_computed(otp, kDifOtpCtrlPartitionSecret2,
+                                               &is_computed));
+  if (is_computed) {
+    uint64_t digest;
+    CHECK_DIF_OK(
+        dif_otp_ctrl_get_digest(otp, kDifOtpCtrlPartitionSecret2, &digest));
+    LOG_INFO("OTP partition locked.");
+    return;
+  }
+  otp_ctrl_testutils_lock_partition(&otp, kDifOtpCtrlPartitionSecret2, 0);
```

Comment:
First argument should be a pointer not a pointer-to-pointer:
```suggestion
  otp_ctrl_testutils_lock_partition(otp, kDifOtpCtrlPartitionSecret2, 0);
```

Reply:
oops! done